### DISCUSSION TITLE
cropping spatial data

### DISCUSSION THREAD

**Discussion by gdauby** [dreamRs/conrappli#28] [General]
Voici le bout de code que j'utilise pour cropper les données spatiales pour les sorties rapport

  ```
  crop_poly <- lapply(rv$spatial_data,
                        function(x)
                          suppressWarnings(st_crop(x = x, y = st_bbox(
                            st_buffer(data_map, 2)
                          ))))
```
Le `suppressWarnings` n'est pas nécessaire.
Le code est valide pour les objets sf, je vais regarder pour le code pour cropper les raster terra.

**Comment by pvictor** [comment]
J'ai cette erreur (sur windows et linux) : 

```r
Error in wk_handle.wk_wkb(wkb, s2_geography_writer(oriented = oriented,  : 
  Loop 2 edge 9 crosses loop 3 edge 0
```

à priori il faut utiliser `sf_use_s2(FALSE)` pour que cela fonctionne.

Le st_buffer() prends un peu de temps, ccette solution pourrait convenir également (conserver que les poly avec un point dedans) : 

```r
pts_dedans <- st_contains(x = protected, y = map_data, sparse = TRUE)
pts_dedans <- apply(pts_dedans, 1, any)
protected <- protected[pts_dedans, ]
```

Sinon pour les raster terra p-e faire ça : 

```r
terra::crop(x = cities, y = terra::ext(st_bbox(map_data)))
```
